modules/admission-webhook-create.adoc

```diff
@@ -0,0 +1,328 @@
+          image: <image_registry_username>/<image_path>:<tag>  <1>
+          imagePullPolicy: IfNotPresent
+          command:
+          - <container_commands>
+          ports:
+          - containerPort: <container_port>
+          volumeMounts:
+          - mountPath: /var/serving-cert
+            name: serving-cert
+          readinessProbe:
+            httpGet:
+              path: /healthz
+              port: <container_port>
+              scheme: HTTPS
+        volumes:
+        - name: serving-cert
+          secret:
+            defaultMode: 420
+            secretName: server-serving-cert
+----
+<1> Points to a webhook server container image.
+
+. Define a secret for the service serving cert signer:
++
+[source,yaml]
+----
+- apiVersion: v1
+  kind: Secret
+  metadata:
+    namespace: <namespace_name>
+    name: server-serving-cert
+  type: kubernetes.io/tls
+  data:  <1>
+    tls.crt: /etc/pki/CA/serving-server.<namespace_name>.svc.crt
+    tls.key: /etc/pki/CA/serving-server.<namespace_name>.svc.key
+----
+<1> References the signed webhook server key and certificate.
+
+. Declare a service account:
++
+[source,yaml]
+----
+- apiVersion: v1
+  kind: ServiceAccount
+  metadata:
+    namespace: <namespace_name>
+    name: server
+----
+
+. Expose the webhook server within the cluster:
++
+[source,yaml]
+----
+- apiVersion: v1
+  kind: Service
+  metadata:
+    namespace: <namespace_name>
+    name: server
+    annotations:
+      service.alpha.openshift.io/serving-cert-secret-name: server-serving-cert
+  spec:
+    selector:
+      server: "true"
+    ports:
+    - port: <host_port>
+      targetPort: <container_port>
+----
+
+. Define a custom resource definition for the webhook server:
++
+[source,yaml]
+----
+- apiVersion: apiextensions.k8s.io/v1beta1
+  kind: CustomResourceDefinition
+  metadata:
+    name: <resource_name>.online.openshift.io  <1>
+  spec:
+    group: online.openshift.io  <2>
+    version: v1alpha1  <3>
+    scope: Cluster  <4>
+    names:
+      plural: <resource_name>  <5>
+      singular: <resource_name_singular>  <6>
+      kind: <camel_case_resource_name>  <7>
+----
+<1> `<plural>.<group>` format and reflects custom resource definition `spec` values.
+<2> REST API group name.
+<3> REST API version name.
+<4> Accepted values are `Namespaced` or `Cluster`.
+<5> Plural name to be included in URL.
+<6> Alias seen in `oc` output.
+<7> Camel case reference for resource manifests.
+
+. Configure the admission webhook server also as an aggregated API server:
++
+[source,yaml]
+----
+- apiVersion: apiregistration.k8s.io/v1beta1
+  kind: APIService
+  metadata:
+    name: v1beta1.admission.online.openshift.io
+  spec:
+    caBundle: /etc/pki/CA/cacert.pem
+    group: admission.online.openshift.io
+    groupPriorityMinimum: 1000
+    versionPriority: 15
+    service:
+      name: server
+      namespace: <namespace_name>
+    version: v1beta1
+----
+
+. Define a webhook, which is in this example validating:
```

Comment:
Suggest "Define a webhook. This example uses a validating webhook."

Reply:
Resolved.